agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -4229,6 +4229,20 @@ bool slave_thread::send_cmdu_to_controller(ieee1905_1::CmduMessageTx &cmdu_tx)
                                   backhaul_params.bridge_mac);
 }
 
+/**
+ * @param[in] None
+ * @return Map of Operating classes and matching supported channels
+ */
+std::map<uint8_t, std::set<uint8_t>> slave_thread::get_supported_channels_map()
+{
+    auto channels_list = hostap_params.supported_channels;
+    std::set<uint8_t> channels_set;
+    for (size_t i = 0; i < beerocks::message::SUPPORTED_CHANNELS_LENGTH; i++) {
+        channels_set.insert(channels_list[i].channel);
+    }
+    return wireless_utils::channel_set_to_operating_classes(channels_set);
```

Comment:
I don't see how why this function is needed.
First, as I have mentioned in another comment, inserting elements to `std::set` is not a good idea from a performance perspective, but it is ok if it happens once. This is not the case here, because you will call it on every time the agent will receive a channel preference query.
I recommend holding a container on the class .h file and update it when the ap_manager joins.
Second, since this function is called only from one place, it can be removed. To shorten things up you could use the constructor of the container to immediately construct the list.
```cpp
  int myints[]= {10,20,30,40,50};
  std::set<int> second (myints,myints+5);        // range
```
This is also supported by `std::list`.
Of course, it will cause to insert zero to the container since `channels_list` is an array with fixed size so it will be a great idea to insert the actual length to the cmdu.